lib/board.rb

```diff
@@ -13,37 +13,35 @@ def add_mark(position, mark)
   end
 
   def available_positions
-    positions = []
-    grid.select { |cell| (0..8).to_a.include? cell}
+    grid.select { |cell| cell != :X && cell != :O}
   end
 
   def game_over?
-    winner || full? ? true : false
+    has_winner? || full?
   end
 
   def winner
     winner = false
     lines.each do |line|
-      winner = :X if all_x(line)
-      winner = :O if all_o(line)
+      return line[0] if all_same_marks?(line)
```

Comment:
to make this more idiomatic but keep the same _semantics_ you could do something like
`lines.find { | line | all_same_marks?(line) }.first`